server/internal/infrastructure/tx-builder/covenantless/builder.go

```diff
@@ -928,15 +934,47 @@ func (b *txBuilder) createPoolTx(
 				}
 			}
 
+			b64, err = ptx.B64Encode()
+			if err != nil {
+				return nil, err
+			}
+
+			feeAmount, err = b.wallet.EstimateFees(ctx, b64)
+			if err != nil {
+				return nil, err
+			}
+
+			if dust > feeAmount {
+				feeAmount = dust
+			} else {
+				feeAmount += dust
+			}
+
+			if dust == 0 {
+				if feeAmount == change {
+					// fees = change, remove change output
+					ptx.UnsignedTx.TxOut = ptx.UnsignedTx.TxOut[:len(ptx.UnsignedTx.TxOut)-1]
+					ptx.Outputs = ptx.Outputs[:len(ptx.Outputs)-1]
+				} else if feeAmount < change {
+					// change covers the fees, reduce change amount
+					ptx.UnsignedTx.TxOut[len(ptx.Outputs)-1].Value = int64(change - feeAmount)
+				} else {
```

Comment:
not sure this is enough: if change - fee amount is dust, we should drop the change amount and put everything on fees, shouldn't we?

Reply:
if yes, the same change must be done below too. Maybe better to move all the common code to a dedicated function?